Package.swift

```diff
@@ -370,3 +384,24 @@ if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == nil {
         .package(path: "../swift-collections"),
     ]
 }
+
+
+#if compiler(>=5.5)
+// SwiftSyntax depends on lib_InternalSwiftSyntaxParser from the toolchain,
+// which had an ABI break in Swift 5.5. As a result, we shouldn't attempt to
+// compile PackageSyntax with an earlier compiler version. Although PackageSyntax
+// should compile with any 5.5 compiler, it will only be functional when built
+// with a toolchain that has a compatible parser library.
+if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == nil {
+    package.dependencies += [.package(url: "https://github.com/apple/swift-syntax.git", .branch("release/5.5"))]
```

Comment:
In the long run, I think it would be better to target `main`/`relatedDependenciesBranch here`. That should ensure that the smoke-testing bot will always use a compatible parser lib and SwiftSyntax, which IMO is the most important configuration to always run tests in. For a self-hosted build, I think it's ok to assume the libraries will sometimes be incompatible and skip the test if so, since that config isn't used for distribution.

Reply:
Agreed — I wasn't able to get that to work with being able to use any version of Xcode to work on SwiftPM at desktop unfortunately (as the newest Xcode (13.0) has Swift 5.5).  So it's not just the self-hosted tests, it's also being able to run the tests locally, where it's less desirable to skip them.